pkg/controller/command/command_upgrade.go

```diff
@@ -30,8 +49,156 @@ func performUpgradeStepIfNeeded(commandCR *contrail.Command, currentDeployment *
 		currentDeployment.Spec.Replicas = commandCR.Spec.CommonConfiguration.Replicas
 		commandCR.Status.UpgradeState = contrail.CommandNotUpgrading
 	}
+	return nil
+}
+
+func (r *ReconcileCommand) deleteMigrationJob(commandCR *contrail.Command) error {
+	job := &batch.Job{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      commandCR.Name + "-upgrade-job",
+			Namespace: commandCR.Namespace,
+		},
+	}
+	pp := meta.DeletePropagationForeground
+	err := r.client.Delete(context.Background(), job, &client.DeleteOptions{PropagationPolicy: &pp})
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	return nil
 }
 
+func getOldCommandImage(list []core.Container, name string) string {
+	for _, container := range list {
+		if name == container.Name {
+			return container.Image
+		}
+	}
+	return ""
+}
+
+func (r *ReconcileCommand) checkDataMigrationJobStatus(command *contrail.Command) error {
+	dataMigrationJob := &batch.Job{}
+	jobName := types.NamespacedName{Namespace: command.Namespace, Name: command.Name + "-upgrade-job"}
+	err := r.client.Get(context.Background(), jobName, dataMigrationJob)
+	alreadyExists := err == nil
+	if alreadyExists {
+		if job.Status(dataMigrationJob.Status).Completed() {
+			command.Status.UpgradeState = contrail.CommandStartingUpgradedDeployment
+		}
+	}
+	if !errors.IsNotFound(err) {
+		return err
+	}
+
+	return r.client.Create(context.Background(), dataMigrationJob)
+}
+
+func (r *ReconcileCommand) reconcileDataMigrationJob(command *contrail.Command, oldImage, newImage, configMapName string) error {
+	dataMigrationJob := &batch.Job{}
+	jobName := types.NamespacedName{Namespace: command.Namespace, Name: command.Name + "-upgrade-job"}
+	err := r.client.Get(context.Background(), jobName, dataMigrationJob)
+	alreadyExists := err == nil
+	if alreadyExists {
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		return err
+	}
+	dataMigrationJob = r.dataMigrationJob(command, oldImage, newImage, configMapName)
+	if err = controllerutil.SetControllerReference(command, dataMigrationJob, r.scheme); err != nil {
+		return err
+	}
+	return r.client.Create(context.Background(), dataMigrationJob)
+}
+
+func (r *ReconcileCommand) dataMigrationJob(commandCR *contrail.Command, oldImage, newImage, configMapName string) *batch.Job {
+	executableMode := int32(0744)
+	return &batch.Job{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      commandCR.Name + "-upgrade-job",
+			Namespace: commandCR.Namespace,
+		},
+		Spec: batch.JobSpec{
+			BackoffLimit: int32ToPtr(5),
+			Template: core.PodTemplateSpec{
+				Spec: core.PodSpec{
+					HostNetwork:   true,
+					RestartPolicy: core.RestartPolicyNever,
+					NodeSelector:  commandCR.Spec.CommonConfiguration.NodeSelector,
+					Volumes: []core.Volume{
+						{
+							Name: configMapName,
```

Comment:
i'd prefer to propagate it from controller since what about situation when someone will change the configuration name in controller and did not know that it is used here, since this config map was created for another job, here I reuse that